From f4e496af66bc6a779f600158cc42b986d654f143 Mon Sep 17 00:00:00 2001
From: Julien Nabet <serval2412@yahoo.fr>
Date: Mon, 14 Dec 2020 19:45:22 +0100
Subject: [PATCH] Related tdf#138888: fix assertion on avmedia/MediaControlBase

Assertion when trying to insert a video in Impress, see bt:
https://bugs.documentfoundation.org/attachment.cgi?id=168161

According to https://en.cppreference.com/w/cpp/algorithm/clamp
"behavior is undefined if the value of lo is greater than hi"

Regression from:
commit 36b3b357fe2a882db6a5f5a006239e16200fb847
Author: Noel <noelgrandin@gmail.com>
Date:   Thu Nov 12 16:10:56 2020 +0200

    replace std::min(std::max()) with std::clamp

Change-Id: Idf33eb6c662b60174678f990f98aefc72edec2a6
Reviewed-on: https://gerrit.libreoffice.org/c/core/+/107722
Tested-by: Jenkins
Reviewed-by: Noel Grandin <noel.grandin@collabora.co.uk>
---
 avmedia/source/framework/MediaControlBase.cxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/avmedia/source/framework/MediaControlBase.cxx b/avmedia/source/framework/MediaControlBase.cxx
index ef92cce8a736..890bb04d805b 100644
--- a/avmedia/source/framework/MediaControlBase.cxx
+++ b/avmedia/source/framework/MediaControlBase.cxx
@@ -64,7 +64,7 @@ void MediaControlBase::UpdateVolumeSlider( MediaItem const & aMediaItem )
     {
         mxVolumeSlider->set_sensitive(true);
         const sal_Int32 nVolumeDB = aMediaItem.getVolumeDB();
-        mxVolumeSlider->set_value( std::clamp( nVolumeDB, sal_Int32(0), AVMEDIA_DB_RANGE ) );
+        mxVolumeSlider->set_value( std::clamp( nVolumeDB, AVMEDIA_DB_RANGE, sal_Int32(0)) );
     }
 }
 
-- 
2.31.1

